web-registry/src/features/marketplace/BuySellOrderFlow.tsx

```diff
@@ -0,0 +1,130 @@
+import { useEffect, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { DeliverTxResponse } from '@cosmjs/stargate';
+
+import { CelebrateIcon } from 'web-components/lib/components/icons/CelebrateIcon';
+import { ProcessingModal } from 'web-components/lib/components/modal/ProcessingModal';
+import { TxErrorModal } from 'web-components/lib/components/modal/TxErrorModal';
+import { Item } from 'web-components/lib/components/modal/TxModal';
+import { TxSuccessfulModal } from 'web-components/lib/components/modal/TxSuccessfulModal';
+
+import { getHashUrl } from 'lib/block-explorer';
+
+import useBuySellOrderSubmit from 'pages/Marketplace/Storefront/hooks/useBuySellOrderSubmit';
+import { VIEW_ECOCREDITS } from 'pages/Projects/Projects.config';
+import { ProjectWithOrderData } from 'pages/Projects/Projects.types';
+import { Link } from 'components/atoms';
+import { BuyCreditsModal } from 'components/organisms';
+import { useMsgClient } from 'hooks';
+
+type Props = {
+  selectedProject: ProjectWithOrderData | null;
+};
+
+export const BuySellOrderFlow = ({ selectedProject }: Props): JSX.Element => {
```

Comment:
This is existing code moved into a reusable component.

Reply:
cool